pkg/operator/ceph/cluster/osd/update.go

```diff
@@ -128,6 +128,16 @@ func (c *updateConfig) updateExistingOSDs(errs *provisionErrors) {
 			continue
 		}
 
+		// backward compatibility for old deployments
+		if osdInfo.DeviceClass == "" {
+			deviceClassInfo, err := cephclient.OSDDeviceClasses(c.cluster.context, c.cluster.clusterInfo, []string{strconv.Itoa(osdID)})
+			if err != nil {
+				errs.addError("failed to update OSD %d. failed to get device class for existing deployment %q. %v", osdID, depName, err)
```

Comment:
Instead of adding an error and skipping the osd update, how about if we just print an error and allow the osd to be updated without the device class? I don't know why the device class wouldn't be available, but it seems like a good case to err on the side of updating the OSD instead of preventing the update.